MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -246,62 +297,251 @@ int main(int argc, char * argv[])
                 }
 
                 std::cout << numPackages << " Package(s) found" << std::endl;
-            }      
+            }
 
             return S_OK;
         }
         case OperationType::Unpack:
         {
             HRESULT hr = S_OK;
 
-            auto packageFilePath = cli.GetPackageFilePathToInstall();
+            auto packageSourcePath = cli.GetPackageFilePathToInstall();
             auto unpackDestination = cli.GetUnpackDestination();
+            auto rootDirectory = cli.GetRootDirectory();
+            WVDFileType fileType = cli.GetFileType();
+            bool createFile = cli.IsCreate();
 
-            if (IsPackageFile(packageFilePath))
+            if (fileType == WVDFileType::CIM)
             {
-                hr = MsixCoreLib::UnpackPackage(packageFilePath, unpackDestination, cli.IsApplyACLs(), cli.IsValidateSignature());
+                if (rootDirectory.empty() || fileType == WVDFileType::NotSpecified)
+                {
+                    std::wcout << std::endl;
+                    std::wcout << "Creating a file with the -create option requires both a -rootDirectory and -fileType" << std::endl;
+                    std::wcout << std::endl;
+                    return E_INVALIDARG;
+                }
+                //if (cli.IsApplyACLs())
+                //{
+                //    std::wcout << std::endl;
+                //    std::wcout << "Applying ACLs is not applicable for CIM files." << std::endl;
+                //    std::wcout << std::endl;
+                //    return E_INVALIDARG;
+                //}
+                if (!EndsWith(unpackDestination, L".cim"))
+                {
+                    std::wcout << std::endl;
+                    std::wcout << "Invalid CIM file name." << std::endl;
+                    std::wcout << std::endl;
+                    return E_INVALIDARG;
+                }
+
+                // Create a temporary directory to unpack package(s) since we cannot unpack to the CIM directly.
+                std::wstring currentDirectory = std::filesystem::current_path();
+                GUID uniqueId;
+                RETURN_IF_FAILED(createGUID(&uniqueId));
+                RPC_STATUS status = UuidCreate(&uniqueId);
+                if (status != RPC_S_OK && status != RPC_S_UUID_LOCAL_ONLY)
+                {
+                    return HRESULT_FROM_WIN32(status);
+                }
+                RPC_WSTR uniqueIDRPCString = NULL;
+                std::wstring uniqueIDString;
+                if (UuidToStringW(&uniqueId, &uniqueIDRPCString) == RPC_S_OK)
+                {
+                    uniqueIDString = (WCHAR*) uniqueIDRPCString;
+                    RpcStringFreeW(&uniqueIDRPCString);
+                }

```

Comment:
maybe your CreateGUID function should also have a CreateGUIDString function if all you really want is the string.